internal/services/resource/resources_data_source.go

```diff
@@ -159,6 +164,12 @@ func filterResource(inputs []resources.GenericResourceExpanded, requiredTags map
 				resID = *res.ID
 			}
 
+			resResourceGroupName := ""
+			if res.ID != nil {
+				re := regexp.MustCompile(`.+/resourceGroups/([^/]+).?`)
+				resResourceGroupName = string(re.FindSubmatch([]byte(*res.ID))[1])
```

Comment:
Rather than doing this, can we use [this function](https://github.com/hashicorp/go-azure-helpers/blob/768f9b56cb76e977dceb971772beb2cf0e09cf6e/resourcemanager/resourceids/legacy.go#L36) to parse the Resource Group from the ID? That'd handle the `resourceGroups` segment differing in casing as required

(note: this does say `legacy` in the name as we're working to remove this parser, but we'll be replacing this with something else, so using the older method is fine here)

Reply:
Hi @tombuildsstuff,

Thanks for the feedback.  I'll take a look at ParseAzureResourceID and post here once I've wrapped my head around it and updated the PR.

Reply:
Hi @tombuildsstuff , I've updated the pull request to use ParseAzureResourceID as requested